File: Data Structures & Algorithms/majority-element/submission-0.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> m1;
        int n = nums.size();
        for (auto i : nums) {
            m1[i]++;
            if (m1[i] > n / 2) {
                return i;
            }
        }
        // According to the problem, a majority element always exists, so this will never be reached.
        return -1;
    }
};

File: Data Structures & Algorithms/rotate-array/submission-0.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        n = len(nums)
        k %= n
        def reversefunc(i:int , j:int)-> None:
            while(i<j):
                nums[i] , nums[j] = nums[j], nums[i]
                i += 1
                j -= 1
        reversefunc(0, n-1)
        reversefunc(0, k-1)
        reversefunc(k, n-1)






        